ic-helpers/src/tokens.rs

```diff
@@ -238,14 +241,19 @@ impl Tokens256 {
     /// Number of bytes needed to represent the value.
     pub const BYTE_LENGTH: usize = 256 / 8;
 
-    /// Converts the value to Tokens128. Returns `None` if the value is greater than `Tokens128::MAX`.
+    /// Converts the value to Tokens128. Returns `None` if the value is greater
+    /// than `Tokens128::MAX`.
     pub fn to_tokens128(&self) -> Option<Tokens128> {
-        let limbs = self.0.limbs();
-        if limbs[2].0 != 0 || limbs[3].0 != 0 {
+        let bytes = self.0.to_le_bytes();
+        let mut num = 0;
+        for (i, &byte) in bytes.iter().enumerate().take(Tokens128::BYTE_LENGTH) {
+            num += byte as u128 * 256u128.pow(i as u32);
+        }
+
+        if bytes[16..].iter().any(|&byte| byte != 0) {
```

Comment:
Replace `16` with `Tokens128::BYTE_LENGTH`, and move the check before the calculation of `num` (we don't need to do all the calculations in case this check fails.